### PR TITLE
Support Swoole

### PR DESCRIPTION
**Pull request by heyanlong**

### CODE REVIEW

**Comment by owenliang** [comment]
Could you optimize all the std::string temporary objects, which have a greater impact on performance in the critical path, thanks 😁

File: src/segment.cc [hunk 1]
```diff
@@ -26,7 +28,10 @@ Segment::Segment(const std::string &serviceId, const std::string &serviceInstanc
     _header = header;
     _version = version;
 
-    std::string traceId = Manager::generateUUID();
+    static std::random_device dev;
+    static std::mt19937 rng(dev());
```
Comment (owenliang) [inline src/segment.cc:32]: init once globally?

File: src/sky_execute.cc [hunk 2]
```diff
@@ -20,21 +21,44 @@
...
 
     zend_function *fn = execute_data->func;
     int is_class = fn->common.scope != nullptr && fn->common.scope->name != nullptr;
     char *class_name = is_class ? ZSTR_VAL(fn->common.scope->name) : nullptr;
     char *function_name = fn->common.function_name != nullptr ? ZSTR_VAL(fn->common.function_name) : nullptr;
+
+    // swoole
+    bool swoole = false;
+    zval *sw_response;
+    if (function_name != nullptr && std::string(function_name) == "{closure}") {
```
Comment (owenliang) [inline src/sky_execute.cc:41]: use strcmp instead?

File: src/sky_execute.cc [hunk 3]
```diff
@@ -20,21 +21,44 @@
...
+        uint32_t arg_count = ZEND_CALL_NUM_ARGS(execute_data);
+        if (arg_count == 2) {
+            zval *sw_request = ZEND_CALL_ARG(execute_data, 1);
+            sw_response = ZEND_CALL_ARG(execute_data, 2);
+            if (Z_TYPE_P(sw_request) == IS_OBJECT && Z_TYPE_P(sw_response) == IS_OBJECT) {
+                if (std::string(Z_OBJ_P(sw_request)->ce->name->val) == "Swoole\\Http\\Request") {
+                    if (std::string(Z_OBJ_P(sw_response)->ce->name->val) == "Swoole\\Http\\Response") {
```
Comment (owenliang) [inline src/sky_execute.cc:48]: && maybe better

File: src/sky_execute.cc [hunk 4]
```diff
@@ -68,6 +97,16 @@ void sky_execute_internal(zend_execute_data *execute_data, zval *return_value) {
         char *class_name = is_class ? ZSTR_VAL(fn->common.scope->name) : nullptr;
         char *function_name = fn->common.function_name != nullptr ? ZSTR_VAL(fn->common.function_name) : nullptr;
 
+        if (class_name != nullptr && function_name != nullptr) {
+            if (std::string(class_name) == "Swoole\\Http\\Response" && std::string(function_name) == "status") {
```
Comment (owenliang) [inline src/sky_execute.cc:101]: strcmp instead

File: src/sky_execute.cc [hunk 5]
```diff
@@ -68,6 +97,16 @@ void sky_execute_internal(zend_execute_data *execute_data, zval *return_value) {
...
+                auto *segment = static_cast<Segment *>(SKYWALKING_G(segment));
+                zval *status = ZEND_CALL_ARG(execute_data, 1);
```
Comment (owenliang) [inline src/sky_execute.cc:103]: safe to check args count

File: src/sky_module.cc [hunk 6]
```diff
@@ -81,51 +81,71 @@ void sky_module_init() {
...
+    if (request != nullptr) {
+        zval *swoole_header = sky_read_property(request, "header", 0);
+        zval *swoole_server = sky_read_property(request, "server", 0);
+
+        if (SKYWALKING_G(version) == 8) {
+            sw = zend_hash_str_find(Z_ARRVAL_P(swoole_header), "sw8", sizeof("sw8") - 1);
+        }
+
+        header = (sw != nullptr ? Z_STRVAL_P(sw) : "");
+        uri = Z_STRVAL_P(zend_hash_str_find(Z_ARRVAL_P(swoole_server), "request_uri", sizeof("request_uri") - 1));
```
Comment (owenliang) [inline src/sky_module.cc:102]: safe if not found?

File: src/sky_module.cc [hunk 7]
```diff
@@ -81,51 +81,71 @@ void sky_module_init() {
...
-        sw = zend_hash_str_find(Z_ARRVAL_P(carrier), "HTTP_SW3", sizeof("HTTP_SW3") - 1);
-    } else if (SKYWALKING_G(version) == 6 || SKYWALKING_G(version) == 7) {
-        sw = zend_hash_str_find(Z_ARRVAL_P(carrier), "HTTP_SW6", sizeof("HTTP_SW6") - 1);
-    } else if (SKYWALKING_G(version) == 8) {
-        sw = zend_hash_str_find(Z_ARRVAL_P(carrier), "HTTP_SW8", sizeof("HTTP_SW8") - 1);
     } else {
-        sw = nullptr;
+        zend_bool jit_initialization = PG(auto_globals_jit);
+
+        if (jit_initialization) {
```
Comment (owenliang) [inline src/sky_module.cc:108]: jit is cool

File: src/sky_module.cc (hunk 6 above)
Comment (heyanlong) [reply src/sky_module.cc:102]: This can be guaranteed by swoole

**Pull request merged**